src/lifecycle/props-init.js

```diff
@@ -93,7 +94,7 @@ function createNativePropertyDefinition(name, opts) {
 
     // Link up the attribute.
     const attributeName = data(this, 'propertyLinks')[name];
-    if (attributeName && !propData.settingAttribute) {
+    if (attributeName && !propData.settingAttribute && valueHasChanged) {
```

Comment:
When Skate syncing attributes and properties, it relies on `syncingAttribute` property of the `propData` (https://github.com/skatejs/skatejs/blob/master/src/api/component.js#L56)

When this situation (same property set) happens, `this.setAttribute(attributeName, serializedValue);` never fires, which means that `syncingAttribute` is never reset, which results in the problem you're fixing.
Consider this, maybe the better solution would be just reset the `syncingAttribute` value here:
https://github.com/skatejs/skatejs/blob/master/src/lifecycle/props-init.js#L102

something like this:
```
this.setAttribute(attributeName, serializedValue);
if (!valueHasChanged) {
    propData.syncingAttribute = false;
}
```